// react-semantify/lib/elements/input.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classGenerator = require('../mixins/classGenerator');

var _classGenerator2 = _interopRequireDefault(_classGenerator);

var _stateSelector = require('../mixins/stateSelector');

var _stateSelector2 = _interopRequireDefault(_stateSelector);

var _unit = require('../commons/unit');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var defaultClassName = 'ui input';

var Input = _react2.default.createClass({
  displayName: 'Input',


  mixins: [_classGenerator2.default, _stateSelector2.default],

  render: function render() {
    var _props = this.props;
    var className = _props.className;

    var other = _objectWithoutProperties(_props, ['className']);

    if (typeof this.props.children != 'undefined') {
      return _react2.default.createElement(
        _unit.Unit,
        _extends({}, other, {
          className: this.getClassName(defaultClassName),
          type: 'div',
          color: 'null',
          loading: this.getLoading(),
          focus: this.getFocus(),
          error: this.getError() }),
        this.props.children
      );
    } else {
      return _react2.default.createElement(
        _unit.Unit,
        {
          className: this.getClassName(defaultClassName),
          type: 'div',
          color: 'null',
          loading: this.getLoading(),
          focus: this.getFocus(),
          error: this.getError() },
        _react2.default.createElement('input', _extends({}, other, {
          placeholder: this.props.placeholder,
          type: this.props.type }))
      );
    }
  }
});

exports.default = Input;
