// react-semantify/lib/views/items.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classGenerator = require('../mixins/classGenerator');

var _classGenerator2 = _interopRequireDefault(_classGenerator);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var defaultClassName = 'ui items';

var Items = _react2.default.createClass({
  displayName: 'Items',


  mixins: [_classGenerator2.default],

  propTypes: {
    className: _react2.default.PropTypes.string,
    type: _react2.default.PropTypes.string
  },

  render: function render() {
    var _props = this.props;
    var type = _props.type;

    var other = _objectWithoutProperties(_props, ['type']);

    if (typeof type !== 'undefined') {
      if (type !== 'link') {
        type = '';
      }
    }

    return _react2.default.createElement(
      'div',
      _extends({}, other, { className: this.getClassName(defaultClassName, type) }),
      this.props.children
    );
  }
});

exports.default = Items;
